arkouda/dataframe.py

```diff
@@ -2706,3 +2708,152 @@ def invert_permutation(perm):
+                     on: str
+                     ) -> DataFrame:
+    """
+    Utilizes the ak.join.inner_join function to return an ak
+    DataFrame object containing only rows that are in both
+    the left and right Dataframes, (based on the "on" param),
+    as well as their associated values.
+
+    Parameters
+    ----------
+    left : DataFrame
+        The Left DataFrame to be joined
+    right : DataFrame
+        The Right DataFrame to be joined
+    on : str
+        The name of the DataFrame column the join is being
+        performed on
+
+    Returns
+    -------
+    ij_ak_df : DataFrame
+        Inner-Joined Arkouda DataFrame
+    """
+
+    ij = inner_join(left[on], right[on])
+
+    left_cols = left.columns.copy()
+    left_cols.remove(on)
+    right_cols = right.columns.copy()
+    right_cols.remove(on)
+
+    new_dict = {on: left[on][ij[0]]}
+
+    for col in left_cols:
+        new_dict[col] = left[col][ij[0]]
+    for col in right_cols:
+        new_dict[col] = right[col][ij[1]]
+
+    ij_ak_df = DataFrame(new_dict)
+
+    return ij_ak_df
+
+
+def right_join_merge(left: DataFrame,
+                     right: DataFrame,
+                     on: str
+                     ) -> DataFrame:
+    """
+    Utilizes the ak.join.inner_join_merge function to return an
+    ak DataFrame object containing all the rows in the right Dataframe,
+    as well as corresponding rows in the left (based on the "on" param),
+    and all of their associated values.
+    Based on pandas merge functionality.
+
+    Parameters
+    ----------
+    left : DataFrame
+        The Left DataFrame to be joined
+    right : DataFrame
+        The Right DataFrame to be joined
+    on : str
+        The name of the DataFrame column the join is being
+        performed on
+
+    Returns
+    -------
+    right_ak_df : DataFrame
+        Right-Joined Arkouda DataFrame
+    """
+
+    left_cols = left.columns.copy()
+    left_cols.remove(on)
+    right_cols = right.columns.copy()
+    right_cols.remove(on)
+
+    in_left = inner_join_merge(left, right, on)
+
+    # Add a try/except statement in case there are no values in right that aren't in left
+    not_in_left = right[find(setdiff1d(right[on], left[on]), right[on])]
+    for col in left_cols:
+        print(col)
+        # Create a nan array for all values not in the left df
+        nan_arr = zeros(len(not_in_left))
+        nan_arr.fill(np.nan)
+        left_col_type = type(in_left[col])
+        if in_left[col].dtype == int:
+            in_left[col] = akcast(in_left[col], akfloat64)
+        else:
+            nan_arr = akcast(nan_arr, in_left[col].dtype)
+
+        try:
+            not_in_left[col] = left_col_type(nan_arr)
+        except TypeError:
+            not_in_left[col] = nan_arr
+
+    right_ak_df = DataFrame.append(in_left, not_in_left)
+
+    return right_ak_df
+
+
+@typechecked
+def merge(
+    left: DataFrame,
+    right: DataFrame,
+    on: str,
```

Comment:
Correct! This would absolutely be next up on the list of enhancements, but I figured with this PR just utilizing the current functionality to create single-column inner/right/left joins would be optimal

Reply:
sweet! the requirement to expand `inner_join` to accept a list of pdarrays has been captured in #2810. We can update `merge` once that's been worked